src/content/1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage.md

```diff
@@ -0,0 +1,227 @@
+        $this->registerHook('actionCustomerGridQueryBuilderModifier') &&
+        $this->registerHook('actionCustomerFormBuilderModifier') &&
+        $this->registerHook('actionAfterCreateCustomerFormHandler') &&
+        $this->registerHook('actionAfterUpdateCustomerFormHandler') &&
+        $this->installTables()
+    ;
+}
+```
+
+{{% notice note %}}
+Hooks `action`**Customer**`GridDefinitionModifier` and `action`**Customer**`GridQueryBuilderModifier` are built
+using grid id which in this case is **Customer**. If You want to use hook for any other grid 
+You can check any definition factory service in `PrestaShop\PrestaShop\Core\Grid\Definition\Factory` to see available grid ids. Grid id is returned by `getId()` method.
+<br />
+<br />
+Hooks `action`**Customer**`FormBuilderModifier`,`actionAfterCreate`**Customer**`FormHandler` and `actionAfterUpdate`**Customer**`FormHandler` unique identifier which
+in this case is **Customer** is retrieved from its form type **CustomerType**. E.g id retrieved from **ManufacturerType** will be **Manufacturer**. Some types might use
+function `getBlockPrefix` to retrieve the unique id
+{{% /notice %}}
+
+### Adding new column to customers grid
+
+#### Extending grid definition and filters
+
+```php
+use PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinitionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ToggleColumn;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+
+/**
+     * Hook allows to modify Customers grid definition.
+     * This hook is a right place to add/remove columns or actions (bulk, grid).
+     *
+     * @param array $params
+     */
+    public function hookActionCustomerGridDefinitionModifier(array $params)
+    {
+        /** @var GridDefinitionInterface $definition */
+        $definition = $params['definition'];
+
+        $translator = $this->getTranslator();
+
+        $definition
+            ->getColumns()
+            ->addAfter(
+                'optin',
+                (new ToggleColumn('is_allowed_for_review'))
+                    ->setName($translator->trans('Allowed for review', [], 'Modules.Ps_DemoCQRSHooksUsage'))
+                    ->setOptions([
+                        'field' => 'is_allowed_for_review',
+                        'primary_field' => 'id_customer',
+                        'route' => 'ps_democqrshooksusage_toggle_is_allowed_for_review',
+                        'route_param_name' => 'customerId',
+                    ])
+            )
+        ;
+
+        $definition->getFilters()->add(
+            (new Filter('is_allowed_for_review', YesAndNoChoiceType::class))
+            ->setAssociatedColumn('is_allowed_for_review')
+        );
+    }
+```
+
+In this sample a new toggable column which determines if the customer is eligable to review products is added just after another column which has id `optin`. The sample code also demonstrates how add new filter.
+
```

Comment:
Please explain that many column classes are already available - ToggleColumn is just one of them, and refer to its definition to explain the required options